daq/stream_monitor.py

```diff
@@ -0,0 +1,164 @@
+"""Utility class to monitor a bunch of input streams and trigger events"""
+
+import logging
+import fcntl
+import os
+
+from select import poll, POLLIN, POLLHUP, POLLNVAL, POLLERR
+
+LOGGER = logging.getLogger('stream')
+
+class StreamMonitor(object):
+    """Monitor set of stream objects"""
+
+    DEFAULT_TIMEOUT_MS = None
+    timeout_ms = None
+    poller = None
+    callbacks = None
+    idle_handler = None
+    loop_hook = None
+    dirty = False
```

Comment:
It's not clear how the class variables are used as there are instance variables as well in the __init__ function